Assets/Tile walking mini game/Scripts/GridManager.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GridManager : MonoBehaviour
+{
+    [SerializeField] private Tile[][] tileGrid;
+
+    [SerializeField] private TileColorManager tileColorManager;
+
+    [SerializeField] private ObstaclesPosition[][] obstacleLayouts;
+
+    public Action WinEvent;
+
+    public Action LoseEvent;
+    public int UntouchedTileCount { get; private set; }
+
+    private void Awake()
+    {
+        PopulateTileGrid();
+
+        CreateObstaclesLayouts();
+    }
+
+    private void Start()
+    {
+        ResetGrid();
+    }
+
+    public void ResetGrid()
+    {
+        foreach (Tile[] tileColumn in tileGrid)
+        {
+            foreach(Tile tile in tileColumn)
+            {
+                tile.TileStatus = TileStatus.untouched;
+            }
+        }
+
+        ObstaclesPosition[] randomlyPickedLayout = obstacleLayouts[UnityEngine.Random.Range(0, obstacleLayouts.Length)];
+
+        UntouchedTileCount = 25 - randomlyPickedLayout.Length;
+
+        AddingObsctacles(randomlyPickedLayout);
+
+        foreach (Tile[] tileColumn in tileGrid)
+        {
+            foreach (Tile tile in tileColumn)
+            {
+                tileColorManager.ChangeTileColorBasedOnStatus(tile);
+            }
+        }
+
+        ArrangingTilesInAGrid();
+
+        UpdateTileStatus(0, 0);
+    }
+
+    private void PopulateTileGrid()
+    {
+        tileGrid = new Tile[5][];
+
+        Tile[] tiles = FindObjectsOfType<Tile>();
+
+        for (int x = 0, i = 0; x < 5; x++)
+        {
+            tileGrid[x] = new Tile[5];
+
+            for (int y = 0; y < 5; y++, i++)
+            {
+                tileGrid[x][y] = tiles[i];

```

Comment:
Instead of finding the tiles at runtime, why not generate them?

Reply:
I now generated them with the SpawnTiles() method